.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -28,10 +28,16 @@ jobs:
 
       - name: ${{ github.workflow }}
         run: |
-          pre-commit autoupdate
-          until pre-commit run --all-files
-          do
-          echo -e "\e[1mRe-running pre-commits...\e[0m"
+          while :; do
+            pre-commit run --all-files && break || {
+              git diff --quiet && {
+                echo One or more pre-commit hooks failed but re-running will not result in changes.
+                exit 1
+              }
+            echo One or more pre-commit hooks failed and there are changes to the repository.
+            echo -e "\e[1mRe-running pre-commit hooks...\e[0m"
+            git add -A
+            }
```

Comment:
@ndevenish or @Anthchirp, if this gets your blessing, I think we're good to go.

Reply:
might work. Give it a try. 🤷

Reply:
Thanks, here goes.